plenum/server/replica.py

```diff
@@ -1631,25 +1631,23 @@ def send(self, msg, stat=None) -> None:
 
     def caught_up_till_pp_seq_no(self, last_caught_up_pp_seq_no):
         self.addToOrdered(self.viewNo, last_caught_up_pp_seq_no)
-        # self._remove_till_caught_up_pp_seq_no(last_caught_up_pp_seq_no)
+        self._remove_till_caught_up_pp_seq_no(last_caught_up_pp_seq_no)
 
     def _remove_till_caught_up_pp_seq_no(self, last_caught_up_pp_seq_no):
```

Comment:
This cleanup should not be required since node will not be participating before catchup, and if there are any 3PC messages remaining before the node started catching up (after a disconnect or a realisation that it is behind), it will be cleared by regular GC mechanism eventually, so there is no harm in doing it here but you dont need to.